service/cloudconfig/v_0_1_0_master_template.go

```diff
@@ -1,123 +1,153 @@
-package configmap
+package cloudconfig
 
 import (
 	"github.com/giantswarm/certificatetpr"
-	k8scloudconfig "github.com/giantswarm/k8scloudconfig"
+	clustertprspec "github.com/giantswarm/clustertpr/spec"
+	"github.com/giantswarm/k8scloudconfig/v_0_1_0"
+	"github.com/giantswarm/kvmtpr"
 	"github.com/giantswarm/microerror"
 )
 
-type MasterExtension struct {
+func v_0_1_0MasterTemplate(customObject kvmtpr.CustomObject, certs certificatetpr.AssetsBundle, node clustertprspec.Node) (string, error) {
+	var err error
+
+	var params v_0_1_0.Params
+	{
+		params.Cluster = customObject.Spec.Cluster
+		params.Extension = &v_0_1_0MasterExtension{
+			certs: certs,
+		}
+		params.Node = node
+	}
+
+	var newCloudConfig *v_0_1_0.CloudConfig
+	{
+		newCloudConfig, err = v_0_1_0.NewCloudConfig(v_0_1_0.MasterTemplate, params)
+		if err != nil {
+			return "", microerror.Mask(err)
+		}
+
+		err = newCloudConfig.ExecuteTemplate()
+		if err != nil {
+			return "", microerror.Mask(err)
+		}
+	}
+
+	return newCloudConfig.Base64(), nil
+}
+
+type v_0_1_0MasterExtension struct {
 	certs certificatetpr.AssetsBundle
 }
 
-func (me *MasterExtension) Files() ([]k8scloudconfig.FileAsset, error) {
-	filesMeta := []k8scloudconfig.FileMetadata{
+func (e *v_0_1_0MasterExtension) Files() ([]v_0_1_0.FileAsset, error) {
```

Comment:
We have to maintain this for each and every version in case we version k8scloudconfig how we do it right now. I think it is good to not only version the string templates but also some go code around it, but it is definitely a bit more work to do this. Opinion?